ibm/resource_ibm_is_instance_action.go

```diff
@@ -0,0 +1,270 @@
+			ValidateFunctionIdentifier: ValidateAllowedStringValue,
+			Type:                       TypeString,
+			Required:                   true,
+			AllowedValues:              instanceActions})
+	ibmISInstanceActionResourceValidator := ResourceValidator{ResourceName: "ibm_is_instance_action", Schema: validateSchema}
+	return &ibmISInstanceActionResourceValidator
+}
+
+func resourceIBMISInstanceActionCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+	instanceId := ""
+	if insId, ok := d.GetOk(isInstanceID); ok {
+		instanceId = insId.(string)
+	}
+
+	actiontypeIntf := d.Get(isInstanceAction)
+	actiontype := actiontypeIntf.(string)
+
+	getinsOptions := &vpcv1.GetInstanceOptions{
+		ID: &instanceId,
+	}
+	instance, response, err := sess.GetInstance(getinsOptions)
+	if err != nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error Getting Instance (%s): %s\n%s", instanceId, err, response)
+	}
+	if (actiontype == "stop" || actiontype == "reboot") && *instance.Status != isInstanceStatusRunning {
+		return fmt.Errorf("Error with stop/reboot action: Cannot invoke stop/reboot action while instance is not in running state")
+	} else if actiontype == "start" && *instance.Status != isInstanceActionStatusStopped {
+		return fmt.Errorf("Error with start action: Cannot invoke start action while instance is not in stopped state")
+	}
+	createinsactoptions := &vpcv1.CreateInstanceActionOptions{
+		InstanceID: &instanceId,
+		Type:       &actiontype,
+	}
+	if instanceActionForceIntf, ok := d.GetOk(isInstanceActionForce); ok {
+		force := instanceActionForceIntf.(bool)
+		createinsactoptions.Force = &force
+	}
+	_, response, err = sess.CreateInstanceAction(createinsactoptions)
+	if err != nil {
+		if response != nil && response.StatusCode == 404 {
+			return nil
+		}
+		return fmt.Errorf("Error Creating Instance Action: %s\n%s", err, response)
+	}
+	if actiontype == "stop" {
+		_, err = isWaitForInstanceActionStop(sess, d.Timeout(schema.TimeoutUpdate), instanceId, d)
+		if err != nil {
+			return err
+		}
+	} else if actiontype == "start" || actiontype == "reboot" {
+		_, err = isWaitForInstanceActionStart(sess, d.Timeout(schema.TimeoutUpdate), instanceId, d)
+		if err != nil {
+			return err
+		}
+	}
+
+	d.SetId(instanceId)
+	return resourceIBMISInstanceActionRead(d, meta)
+}
+
+func resourceIBMISInstanceActionRead(d *schema.ResourceData, meta interface{}) error {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+	id := d.Id()
+
+	options := &vpcv1.GetInstanceOptions{
+		ID: &id,
+	}
+	instance, response, err := sess.GetInstance(options)
+	if err != nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error getting instance (%s): %s\n%s", id, err, response)
+	}
+	d.SetId(*instance.ID)
+	d.Set(isInstanceStatus, *instance.Status)
```

Comment:
action and action force are not set in READ method

Reply:
these are input parameters and we don't get these from API response. is it needed to be set again reading from the input itself ?